api/src/main/java/org/commonjava/indy/content/GroupRepositoryFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.commonjava.indy.content;
+
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.Group;
+
+import java.util.List;
+
+public interface GroupRepositoryFilter
+                extends Comparable<GroupRepositoryFilter>
+{
+    /**
+     * Higher priority filter get executed first.
+     * @return priority
+     */
+    int getPriority();
+
+    boolean canProcess( String path, Group group );
+
+    /**
+     * Get all concrete stores that may contain the target path.
+     *
+     * @param path
+     * @param group
+     * @param concreteStores in target group ( include stores in sub groups )
+     * @return
+     */
+    List<ArtifactStore> filter( String path, Group group, List<ArtifactStore> concreteStores );
+
+    /**
+     * Get all concrete stores that may contain the first occurrence of the target path.
+     *
+     * @param path
+     * @param group
+     * @param concreteStores in target group ( include stores in sub groups )
+     * @return
+     */
+    List<ArtifactStore> filterForFirstMatch( String path, Group group, List<ArtifactStore> concreteStores );
```

Comment:
How are these two different from one another? It seems like they're used the same way...Can we consolidate down to one method?

Reply:
This is an immature thought when I wanted to further limit the range of repos for the find-first case. But it turnes to be useless. I will remove it.